tests/migration/eviction_strategy.go

```diff
@@ -0,0 +1,720 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package migration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	kvpointer "kubevirt.io/kubevirt/pkg/pointer"
+
+	"kubevirt.io/kubevirt/tests/testsuite"
+
+	"kubevirt.io/kubevirt/tests/clientcmd"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+	"kubevirt.io/kubevirt/tests/framework/cleanup"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libnet"
+	"kubevirt.io/kubevirt/tests/libnode"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/util"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	k8sv1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"kubevirt.io/kubevirt/tests/libwait"
+
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+)
+
+var _ = SIGMigrationDescribe("Live Migration", func() {
+	var (
+		virtClient kubecli.KubevirtClient
+		err        error
+	)
+
+	BeforeEach(func() {
+		checks.SkipIfMigrationIsNotPossible()
+		virtClient = kubevirt.Client()
+	})
+
+	setControlPlaneUnschedulable := func(mode bool) {
```

Comment:
nit: not sure, but that might go into `libinfra`. WDYT?

Reply:
side note: this should only be allowed in `Serial` mode, no? Maybe add an assertion for that?

Reply:
Thanks Daniel. Instead of `libinfra` I opted for `libnode`. We already have `SetNodeUnschedulable` and `SetNodeSchedulable` there and they could be used. In general, I think that this is a function that for the time being, it is not worth generalizing. I have extracted a general `libnode.GetControlPlaneNodes` function, but left the wrapper `setControlPlaneSchedulability` function here. WDYT? Thanks